T18074 View to show test plans for each kernel revision

alexandrasp opened the pull request:
View test plans for each kernel revision.

Commit:
add new test plan view for each kernel revision

update kernel ci front-end in order to create new view
for test plans from a kernel revision combined with
a branch and a kernel version.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

Commit:
update routes to redirect to test plan view

create links to new view where you can
see test plans filtered by kernel, branch and tree.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

Commit:
update require js with minimal version

it was updated to use minimal version of require.js for
code optimizations.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

alexandrasp commented on app/dashboard/static/js/app/view-test-plan-all.2020.1.20.js:
```diff
@@ -0,0 +1,532 @@
+        data = {
+            tree: gTree,
+            git_branch: gBranch,
+            //distinct: 'name',
```
@gctucker seems that this query filtering by `kernel`, `branch` and `tree` don't match anything. Is distinct available for `test_group`? It is important for test plan list..because we need to generate unique ids to update the table in order to put the numbers.

gctucker replied:
See how I did this in the view I made:
https://github.com/gctucker/kernelci-frontend/blob/T18074-test-plans-view/app/dashboard/static/js/app/view-tests-job-branch-kernel.2020.1.js#L279

alexandrasp replied:
sure, you aggregated the data :)

Commit:
update query to filter test plan

process queries on mongodb filtering by kernel, branch, tree
and unique test plans, as well as update the code to work with
new responses format.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

Commit:
update view to work on case of no test plan data

update to do the page work in cases that there are not any test case
to show.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

Commit:
update template

remove all unused fields from the view template.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>
